app/_hub/kong-inc/oauth2-introspection/index.md

```diff
@@ -46,88 +47,101 @@ params:
     - name: timeout
       required: false
       default: 10000
       value_in_examples:
+      datatype: integer
       description: |
-        An optional timeout in milliseconds when sending data to the upstream server
+        An optional timeout in milliseconds when sending data to the upstream server.
     - name: keepalive
       required: false
       default: 60000
       value_in_examples:
+      datatype: integer
       description: |
-        An optional value in milliseconds that defines for how long an idle connection will live before being closed
+        An optional value in milliseconds that defines how long an idle connection lives before being closed.
     - name: anonymous
       required: false
       default:
       value_in_examples:
+      datatype: string
       description: |
         An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure 4xx.
     - name: run_on_preflight
       required: false
       default: true
       value_in_examples:
+      datatype: boolean
       description: |
-        A boolean value that indicates whether the plugin should run (and try to authenticate) on `OPTIONS` preflight requests. If set to `false` then `OPTIONS` requests will always be allowed.
+        A boolean value that indicates whether the plugin should run (and try to authenticate) on `OPTIONS` preflight requests. If set to `false`, then `OPTIONS` requests will always be allowed.
     - name: consumer_by
-      required: false
-      default: username
+      required: true
+      default: CONSUMER_BY_DEFAULT
       value_in_examples: username
+      datatype: string
       description: |
-        A string indicating whether to associate oauth2 `username` or `client_id`
-        with the consumer's username. Oauth2 `username` is mapped to a consumer's
-        `username` field, while an oauth2 `client_id` maps to a consumer's
-        `custom_id`
+        A string indicating whether to associate OAuth2 `username` or `client_id`
+        with the consumer's username. OAuth2 `username` is mapped to a consumer's
+        `username` field, while an OAuth2 `client_id` maps to a consumer's
+        `custom_id`.
     - name: introspect_request
-      required: false
+      required: true
       default: false
       value_in_examples:
+      datatype: boolean
       description: |
         A boolean indicating whether to forward information about the current
         downstream request to the introspect endpoint. If true, headers
-        `X-Request-Path` and `X-Request-Http-Method` will be inserted in the
-        introspect request
+        `X-Request-Path` and `X-Request-Http-Method` will be inserted into the
+        introspect request.
     - name: custom_introspection_headers
-      required: false
+      required: true
       default:
       value_in_examples:
+      datatype: map of string values
```

Comment:
```suggestion
      datatype: map of string keys and string values
```

Can we add examples too? If I'm an user and see this datatype I'd love some examples at least in JSON.

Reply:
Of course we can add them, but can you please provide them? So this request is scope creep beyond adding datatypes...